src/bytes_mut.rs

```diff
@@ -1283,9 +1283,7 @@ impl Extend<u8> for BytesMut {
 
         // TODO: optimize
         // 1. If self.kind() == KIND_VEC, use Vec::extend
-        // 2. Make `reserve` inline-able
```

Comment:
I'm not sure what this TODO comment originally meant, but if it means something like "reserve (in here or put_u8) should be able to omit if the compiler has knowledge of the actual length of iter" it may not have been resolved yet.

([In my experience](https://github.com/openrr/mesh-loader/blob/2405ef08511c595fa8a754b08e15a0be32513d67/src/stl/mod.rs#L185-L187), the compiler optimizations around the reserve have not worked that well.)

Reply:
Thanks @taiki-e! I think it refers to https://github.com/tokio-rs/bytes/pull/313, which was just after the introduction of the comment in https://github.com/tokio-rs/bytes/pull/305. I should've mentioned that in my commit though. Thanks for taking a look!